src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -63,25 +60,58 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Wire-up static web assets -->
   <PropertyGroup>
-    <GenerateStaticWebAssetsPublishManifestDependsOn Condition="'$(WasmBuildingForNestedPublish)' != 'true'">
-      $(GenerateStaticWebAssetsPublishManifestDependsOn);
-      _BlazorCompressPublishFiles;
-    </GenerateStaticWebAssetsPublishManifestDependsOn>
+    <ResolveCompressedFilesDependsOn>
+      $(ResolveCompressedFilesDependsOn);
+      _ResolveBlazorWasmOutputs;
+    </ResolveCompressedFilesDependsOn>
+
+    <ResolveCompressedFilesForPublishDependsOn>
+      $(ResolveCompressedFilesForPublishDependsOn);
+      _CollectAssetsToCompressForPublish;
+    </ResolveCompressedFilesForPublishDependsOn>
 
     <AddWasmStaticWebAssetsDependsOn>
       $(AddWasmStaticWebAssetsDependsOn);
       _ResolveBlazorWasmOutputs;
     </AddWasmStaticWebAssetsDependsOn>
+
     <GenerateBuildWasmBootJsonDependsOn>
       $(GenerateBuildWasmBootJsonDependsOn);
       GenerateBlazorBootExtensionJson;
     </GenerateBuildWasmBootJsonDependsOn>
+
     <GeneratePublishWasmBootJsonDependsOn>
       $(GeneratePublishWasmBootJsonDependsOn);
       GenerateBlazorBootExtensionJson;
     </GeneratePublishWasmBootJsonDependsOn>
   </PropertyGroup>
 
+  <ItemGroup>
+
+    <!--
+      Compress referenced binaries using GZip during build. This skips files such as the project's assemblies
+      that change from build to build. Runtime assets contribute to the bulk of the download size. Compressing it
+      has the most benefit while avoiding any ongoing costs to the dev inner loop.
+    -->
+    <CompressionConfiguration Include="BlazorBuildGzip">
```

Comment:
For the compression configuration we should have "Build, Publish, All". With "All" meaning that it runs for the pattern during build and publish. For example, we want to gzip all the bits during build, but we also want to gzip them in publish, and there's no value in having to specify two separate configuration entries.